// ember-handlebars-brunch/lib/ember-handlebars-compiler.js
// Generated by CoffeeScript 1.4.0
(function() {

  module.exports = (function() {
    var compilerPath, compilerjs, context, element, fs, sandbox, sysPath, vm;
    fs = require('fs');
    vm = require('vm');
    sysPath = require('path');
    compilerPath = sysPath.join(__dirname, '..', 'vendor', 'ember-template-compiler.js');
    compilerjs = fs.readFileSync(compilerPath, 'utf8');
    element = {
      firstChild: function() {
        return element;
      },
      innerHTML: function() {
        return element;
      }
    };
    sandbox = {
      document: {
        createRange: false,
        createElement: function() {
          return element;
        }
      },
      console: console,
      template: null,
      templatejs: null,
      exports: {
        precompile: null
      }
    };
    sandbox.window = sandbox;
    context = vm.createContext(sandbox);
    vm.runInContext(compilerjs, context, 'compiler.js');
    return function(templateData) {
      context.template = templateData;
      vm.runInContext('templatejs = exports.precompile(template).toString();', context);
      return context.templatejs;
    };
  })();

}).call(this);
